djangoProject3/my-bookstore/src/components/Registration.js

```diff
@@ -0,0 +1,127 @@
+import React, {useState} from 'react';
+import axios from 'axios';
+import {Link} from 'react-router-dom';
+import NavBar from './Navbar';
+import InputField from './inputField';
+
+const Register = () => {
+    const [formData, setFormData] = useState({
+        username: '',
+        password: '',
+        password2: '',
+        email: '',
+        first_name: '',
+        last_name: '',
+    });
+
+    const [errorMessage, setErrorMessage] = useState('');
+    const [successMessage, setSuccessMessage] = useState('');
+    const [token, setToken] = useState('');
+    const [redirectToNavbar, setRedirectToNavbar] = useState(false);
+
+    const handleChange = (e) => {
+        setFormData({
+            ...formData,
+            [e.target.name]: e.target.value,
+        });
+    };
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        setErrorMessage('');
+        setSuccessMessage('');
+
+        try {
+            const response = await axios.post('http://localhost:8000/register/', formData);
+            sessionStorage.setItem('authToken', response.data.token);
+            sessionStorage.setItem('username', formData.username);
+            setToken(response.data.token);
+            setSuccessMessage('Registration successful');
+            setRedirectToNavbar(true);
+        } catch (errorMessage) {
+            if (errorMessage.response) {
+                const errorData = errorMessage.response.data;
+                if (errorData.errors) {
+                    setErrorMessage(Object.values(errorData.errors)[0]); // Display the first error
+                } else if (errorData.message) {
+                    setErrorMessage(errorData.message);
+                }
+            } else {
+                setErrorMessage('No Server Response');
+            }
+        }
+    };
+
+    return (
+        <>
+            {redirectToNavbar ? <NavBar/> : null}
+            <div className="container login-container my-3">
+                {successMessage && (
+                    <div>
+                        <h3 className="text-success">{successMessage}</h3>
+                    </div>
+                )}
+                {errorMessage && (
+                    <div className="alert alert-danger" role="alert">
+                        <p>{errorMessage}</p>
+                    </div>
+                )}
+                <h2>Register</h2>
+               <form onSubmit={handleSubmit}>
+                    <InputField
+                        label="Username"
```

Comment:
you can create an array of objects containing ```label, type, name...``` and then map on it to reduce code duplication